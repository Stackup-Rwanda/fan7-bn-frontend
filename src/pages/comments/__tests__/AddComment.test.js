import React from 'react';
import { mount , shallow } from 'enzyme';
import thunk from 'redux-thunk';
import { Provider } from 'react-redux';
import ConfigureStore from 'redux-mock-store';
import { BrowserRouter as Router } from 'react-router-dom';
import Comment from '../index';

const middlewares = [thunk];
const mockStore = ConfigureStore(middlewares);

const history = {
    push: jest.fn(),
    replace: jest.fn(),
    createHref: jest.fn(),
};

describe('Comment Component Test', () => {
    it('should return true if component exists', () => {
        const initialState = {};
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                <Comment />
                </Router>
            </Provider>
        );

        expect(wrapper.find('.request_container').exists()).toBe(true);
    });
    
    it('should display the content if it is not loading', () => {
        const initialState = {
            addComment: { isLoading:false, comments: [], request: [], comment: '' },
        };
        const store = mockStore(initialState);
        const wrapper = mount(
            <Provider store={store}>
                <Router history={history}>
                    <Comment />
                </Router>
            </Provider>,
        );
        expect(wrapper.find('.request_container_request').exists()).toBe(true);
        expect(wrapper.find('.request_container_comment').exists()).toBe(true);
        
    });
});

describe('<Comment /> shallow rendering tests', () => {
    const initialState = {};
    const store = mockStore(initialState);

    it('matches the snapshot', () => {
        const tree = shallow(<Comment store={store} />);
        expect(tree).toMatchSnapshot();
    });
});