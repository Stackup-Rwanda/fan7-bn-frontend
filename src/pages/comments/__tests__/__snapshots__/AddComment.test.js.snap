// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Comment /> shallow rendering tests matches the snapshot 1`] = `
<ContextProvider
  value={null}
>
  <Comment
    getComments={[Function]}
    getRequest={[Function]}
    postComment={[Function]}
    removeComment={[Function]}
    store={
      Object {
        "clearActions": [Function],
        "dispatch": [Function],
        "getActions": [Function],
        "getState": [Function],
        "replaceReducer": [Function],
        "subscribe": [Function],
      }
    }
  />
</ContextProvider>
`;
